app/scripts/views/search/location-search-directive.js

```diff
@@ -23,9 +24,15 @@
 
             ctl.clearError = clearError;
             ctl.onSearchButtonClicked = onSearchButtonClicked;
+            ctl.onTextKeyup = onTextKeyup;
         }
 
         function search(searchText, magicKey, options) {
+            if (searching) {
```

Comment:
Hmm, yeah the only case where this is used is when the user hits enter with a typeahead option selected. In that case, the typeahead-on-selected function is called as well as the ng-keyup, so this just prevents search from being called twice. I'll add a comment detailing this and a note to switch if it becomes a problem.